### ISSUE TITLE
Plumbing and electricity

### ISSUE THREAD

**Issue by IriaSomobu** [cataclysmbnteam/Cataclysm-BN#48]
Maybe this feature is too broad and complex for now, but it'll be very handy to have building system similar to one used for cars.

The problem is, you cannot build plumbing or electric network on your base, so you cannot properly use boilers, sinks, fridges etc.

The solution I'm thinking of is to implement building menu. In that menu player should be able to
- connect electricity sources (solar panels, generators) to consumers (lamps, fridges, boilers, water pumps) within one window;
- connect water sources (water pumps) to sinks (boiler, tank, water sink) with water pipe tiles visible only in building mode.

*N.B. I can draw illustration how it should work if this part is unclear*

Also existing structures (houses, labs) can be updated with this feature. In this case it'll be easier for player to use already-built house as base.

To sum up, this mechanic has low impact on core (rogue-like) gameplay, but makes interior objects more useful, improves QoL on 'motherbase' and gives additional depth to building.

**Comment by Igryn** [comment]
Would be nice. Current house building system kind of laggs behind vehicle constructing which has way much more variety and flexibility

**Comment by Coolthulhu** [comment]
The water pipe system sounds like a relatively complex thing that would delay it a fair bit.
I assume you want it primarily to separate liquid networks, so that clean water doesn't mix with dirty water.
The pipe system would need to be cached - "crawling" the pipes on every water use would be expensive, especially if the pipes were long. Calculating pumped water would be prohibitively costly without some good design to skip calculations.

I see a simpler solution, mirroring the electric network:
* An overmap tile can have arbitrary number of liquid networks
* Liquid network component can be connected to any liquid network on the overmap tile
* Two adjacent networks can be connected into one if they use the same liquid
* Extending a network over a tile that it doesn't cover requires a build job
* Creating a new network (for example, for a new liquid) requires a build job
* If a network has no liquid, it can be switched into a new liquid type

This implementation would be much simpler and could share most of the code with electric networks.
It would also require very little extra effort from the player - no need to drag pipes between elements, just do "connect to network" job on a sink and that's it.

**Comment by Coolthulhu** [comment]
In plans. I assume I'll be the one implementing it. If someone else wants to try, I could create a document outlining a reasonable, maintainable design.
